tests/unit/tool/test_grype.py

```diff
@@ -21,3 +21,19 @@ def test_grype_no_profile():
         tool = Grype(path="test-path")
         tool.capture(image="test-image", tool_input=None)
         assert check_output.call_args.args[0] == ["test-path/grype", "-o", "json", "test-image"]
+
+
+def test_install_from_path():
+    with mock.patch("subprocess.check_call") as check_call, mock.patch("git.Repo") as repo, mock.patch(
+        "os.path.exists"
+    ) as exists, mock.patch("os.makedirs") as makedirs, mock.patch("os.chmod") as chmod:
+        check_call.return_value = bytes("test-output", "utf-8")
+        exists.return_value = True
+        fake_repo = mock.Mock()
+        fake_repo.git = mock.Mock()
+        fake_repo.git.describe.return_value = "test-version"
+        repo.return_value = fake_repo
+        tool = Grype.install(
+            version="path:/where/grype/is/cloned", path=".yardstick/tools/grype/path:_where_grype_is_cloned", update_db=False
```

Comment:
this seems a little confusing (there is an encoding of `path:` within a path within the `path` variable). Also, one thing that's important is to make certain that subsequent runs wont try an reuse the same (already installed) binary just because the path matches... in case you make code changes and try and run this again. There is some of this logic in other places, but I don't see it in this execution path, so you're probably fine (but wanted to mention it just in case).

Lastly, maybe instead of the full path within `path` taking the sha256sum of the path string to get a single directory would flatten out where it is stored within the tools directory some (I could be misunderstanding something here though, but it looks like it could make a deeply nested path here).

Reply:
Other code in yardstick normalizes the path by replacing `/` with `_`: https://github.com/anchore/yardstick/blob/09fbfdafa768de0bff699cbe8c3615ff601201b5/src/yardstick/store/tool.py#L15-L20

So this ends up with a potentially long path, but there's no additional nesting.

For example, where I've been using this in grype:

```
❯ ls -1 .yardstick/tools/grype
_Users_willmurphy_work_grype
v0.65.1
```

I think keeping the path human-readable is important, since it enables comparing two different local checkouts of grype and being able to tell at a glance which is which.

The concern that a subsequent run with a changed grype at the same path is valid. Right now, `_install_from_path` builds grype unconditionally, so I don't think anything should go wrong there.

Reply:
good point -- we are already normalizing these in the store 🎉

Reply:
from an offline conversation we think that `path:<the-path>-<git description w/dirty>[-<git diff hash if dirty>]` will work pretty well here as the resolved version value.